### PR TITLE
Add test to validate elasticsearch-operator image

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

**Comment by kevinearls** [comment]
@objectiser @jpkrohling @pavolloffay Please review

**Comment by pavolloffay** [comment]
@kevinearls why is this needed? Can't we just rely on functional tests?

**Comment by kevinearls** [comment]
@pavolloffay This is for some CI jobs that I'm working on where we will test against different versions of the elasticsearch-operator.  It's not strictly needed, but is just to confirm that the elasticsearch-operator is actually deployed with the image that we think it's using.  That can't really be done with functional tests.

File: test/e2e/self_provisioned_elasticsearch_test.go [hunk 1]
```diff
@@ -95,6 +97,17 @@ func (suite *SelfProvisionedTestSuite) TestSelfProvisionedESSmokeTest() {
 	require.NoError(t, err, "Error running smoketest")
 }
 
+func (suite *SelfProvisionedTestSuite) TestValidateEsOperatorImage() {
```
Comment (pavolloffay) [inline test/e2e/self_provisioned_elasticsearch_test.go]: Is this being called? I didn't find from where it is invoked.

Comment (kevinearls) [reply test/e2e/self_provisioned_elasticsearch_test.go]: That is the standard test signature when using the [stretchr/testify suite](https://github.com/stretchr/testify) package

Comment (pavolloffay) [reply test/e2e/self_provisioned_elasticsearch_test.go]: ah ok I thought that will will not be automatically  invoked.

**Pull request merged**